### DISCUSSION TITLE
S Pen switch to Eraser on Tab S6 Lite

### DISCUSSION THREAD

**Discussion by r0gueone** [saber-notes/saber#396] [Q&A] [answered]
I have been testing Saber with my Samsung Tab S6 Lite, and although the handwriting works very well, pressing the S Pen button doesn't do anything.  I have to manually choose the Eraser option to erase.  Is there a way to remap the control to it?  I see tooltip info about using CTRL+E on Windows to do it, but that obviously doesn't work on Android.

I love the NextCloud integration and I want to use it over S Notes, but having to manually choose the eraser, and then manually chose the pen again is cumbersome.

**Comment by adil192** [comment]
Do you experience https://github.com/flutter/flutter/issues/111068#issuecomment-1315138150?
> I can confirm that the app no longer crashes when I touch the touchscreen with the stylus and hold down the button. +1
However, the pen then seems to disappear completely for the app and is no longer recognised at all. This means that no PointerDown, PointerMove, PointerUp or similar events are triggered when the button is pressed. This also means that no onTap, onPressed etc. reacts.

**Reply by adil192** [reply]
Also this is a Flutter bug that I can't do much about. Styluses other than the S Pen seem to work. See https://github.com/adil192/saber/issues/21 for the relevant issue

**Reply by r0gueone** [reply]
No, I never experienced that.  I've only had the Tab S6 for a little over a month, and when I tried Saber it never crashed for me, the S Pen button just never did anything.

**Reply by adil192** [reply]
The crash was fixed back in Flutter 3.7 (Saber v0.6.0), do you experience the disappearing behaviour? i.e. When you press the stylus button, does it act like a pen or is the stylus ignored entirely?

**Reply by r0gueone** [reply]
Hovering the S Pen over the table screen gives me a small circle pointer.  If I press the button and hold it, the circle is there until I touch the screen.  Then the pointer goes away.  Releasing the button brings back the pointer circle.  A single click of the S Pen button doesn't do anything, unless I have the S Pen settings set to bring up the air command dialog.  Disabling air command dialog, and just a single click in Saber again does nothing

**Comment by ceskyDJ** [comment]
I have similar experience. It doesn't crash (as in earlier versions), when I use the button on SPen but it does not work as an eraser.

**Comment by WasteOfO2** [comment, marked as answer]
This is an upstream issue https://github.com/flutter/flutter/issues/42846
